hidden/Features/StatusBar/MouseMovementHandler.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  MouseMovementHandler.swift
+//  Hidden Bar
+//
+//  Created by Peter Luo on 2021/3/4.
+//  Copyright © 2021 Dwarves Foundation. All rights reserved.
+//
+
+import AppKit
+
+class MouseMovementHandler {
+    
+    var expandHandler   : () -> Void
+    var collapseHandler : () -> Void
+    
+    private var mouseMoveMoniter : GlobalEventMoniter?
+    private var mouseClickMoniter : GlobalEventMoniter?
+    
+    private var interactedWithMenuBar = false
+    private var mouseIsOnMenuBar      = false {
+        didSet {
+            guard mouseIsOnMenuBar != oldValue else { return }
+            if mouseIsOnMenuBar {
+                interactedWithMenuBar = false
+                expandHandler()
+            } else if !interactedWithMenuBar {
+                collapseHandler()
+            }
+        }
+    }
+    
+    init(expandHandler: @escaping ()->Void, collapseHandler: @escaping ()->Void) {
+        self.expandHandler = expandHandler
+        self.collapseHandler = collapseHandler
```

Comment:
I don't understand. `collapseHandler` and `expandHandler` is both needed for `MouseMovementHandler`. Do you mean removing both of them and use other ways to expand/collapse?

Reply:
Because if you press on any app status bar to open they menu, then you want to select menu, at that point your mouse cursor move out of status bar frame cause the opened menu disappear.

Reply:
If you left click inside the menubar it won't collapse after the mouse leaves the menu bar.
But now that you mentioned this I realized I didn't take right click into account.

Reply:
OK, I know what you mean. `NSEvent` GlobalMoniter doesn't seem to work when the app is the first responder.
Is there a way to get notified when app menu is shown? That way we can disable auto collapse when the app menu is in use.

Reply:
Also for the same reason, auto expand/collapse feature doesn't work when the preferences window is on top.

Reply:
Let me try my solution, I'm not sure it work or not, I will let you know later.